lib/src/common/managed_stream_subscription.dart

```diff
@@ -0,0 +1,42 @@
+import 'dart:async';
+
+/// An implementation of StreamSubscription that provides a didCancel future.
+class ManagedStreamSubscription<T> extends StreamSubscription<T> {
```

Comment:
No reason. The implementation isn't publicly exported so the two are equivalent.